package/gluon-web-wifi-config/luasrc/lib/gluon/config-mode/model/admin/wifi-config.lua

```diff
@@ -95,9 +95,13 @@ uci:foreach('wireless', 'wifi-device', function(config)
 
 	vif_option('client', {'client', 'owe'}, translate('Enable client network (access point)'))
 
-	local mesh_vif = vif_option('mesh', {'mesh'}, translate("Enable mesh network (802.11s)"))
-	if is_5ghz then
-		table.insert(mesh_vifs_5ghz, mesh_vif)
+	-- disable setting mesh iface setting if radio does not support configured mesh channel
+	if wireless.supports_channel(config, tonumber(config.channel)) then
+		local mesh_vif = vif_option('mesh', {'mesh'}, translate("Enable mesh network (802.11s)"))
+
+		if is_5ghz then
+			table.insert(mesh_vifs_5ghz, mesh_vif)
+		end
```

Comment:
Maybe add a notice that it's unsupported instead of just removing the checkbox?

Reply:
I don't quite now how this should be done with our config mode. So this is still open

Reply:
See https://github.com/freifunk-gluon/gluon/blob/main/package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua#L29-L35 for a conditionally displayed warning. This refers to the template in gluon-web-model/files/lib/gluon/web/view/model/warning.html - a similar template could be added for a "notice".

Reply:
Thanks, then it would just look like this:
![image](https://github.com/user-attachments/assets/16a53614-a179-4a57-a4b5-4ae771189000)

What do you think of it @neocturne ? :)

Reply:
I added a notice as you suggested now. I think this looks better:

![image](https://github.com/user-attachments/assets/a063686d-ecb5-4bbc-8cd9-b754ec88e1f5)